apis/hive/v1/hiveconfig_types.go

```diff
@@ -144,8 +143,45 @@ type HiveConfigSpec struct {
 	// extract metrics. The operator also sets up RBAC in the TargetNamespace so that openshift
 	// prometheus in the cluster can list/access objects required to pull metrics.
 	ExportMetrics bool `json:"exportMetrics,omitempty"`
+
+	// MetricsConfig encapsulates metrics specific configurations, like signing up for opt-in metrics
+	// +optional
+	MetricsConfig *MetricsConfig `json:"metricsConfig,omitempty"`
+}
+
+type MetricsConfig struct {
+	// Optional metrics and their configurations
+	// +optional
+	MetricsWithDuration []MetricsWithDuration `json:"metricsWithDuration"`
 }
 
+// MetricsWithDuration represent metrics that report time as values,like transition seconds
+type MetricsWithDuration struct {
+	// Name of the metric. It will correspond to an optional relevant metric in hive
+	Name DurationMetricType `json:"name"`
+	// Duration is the minimum time taken - the relevant metric will be logged only if the value reported by that metric
+	// is more than the time mentioned here. For example, if a user opts-in for current clusters stopping and mentions
+	// 1 hour here, only the clusters stopping for more than an hour will be reported
+	// This is a Duration value; see https://pkg.go.dev/time#ParseDuration for accepted formats.
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$"
+	Duration *metav1.Duration `json:"duration"`
+}
+
+// DurationMetricType is a valid value for MetricsWithDuration.Name
+// +kubebuilder:validation:Enum=CurrentStopping;CurrentResuming;CurrentWaitingForCO;CumulativeHibernated;CumulativeResumed
+type DurationMetricType string
+
+const (
+	CurrentStopping     DurationMetricType = "CurrentStopping"
+	CurrentResuming     DurationMetricType = "CurrentResuming"
+	CurrentWaitingForCO DurationMetricType = "CurrentWaitingForCO"
+
+	// These metrics will not be cleared and can potentially blow up the cardinality
+	CumulativeHibernated DurationMetricType = "CumulativeHibernated"
+	CumulativeResumed    DurationMetricType = "CumulativeResumed"
+)
```

Comment:
This is purely done for the sake of mapping in metrics controller, and serves no other purpose. The real info for a use trying to set this value comes from the enum validation on DurationMetricType.